src/lib/explorer/extensions/commands/addnode.cpp

```diff
@@ -34,7 +36,29 @@ namespace commands {
 console_result addnode::invoke (Json::Value& jv_output,
          libbitcoin::server::server_node& node)
 {
-    jv_output["message"] = IN_DEVELOPING;
+    //jv_output["message"] = IN_DEVELOPING;
+
+    administrator_required_checker(node, auth_.name, auth_.auth);
+    auto& root = jv_output;
+
+    auto address = libbitcoin::config::authority(argument_.address).to_network_address();
+
+    code errcode;
+    auto handler = [&errcode](const code& ec){
+        errcode = ec;
+    };
+
+    if (argument_.command == "add") {
+        node.store(address, handler);
+    } else if (argument_.command == "remove") {
+        node.remove(address, handler);
+    } else {
+        jv_output["message"] = std::string("invalid command[") + argument_.command + "].";
+        return console_result::okay;
+    }
+
+    jv_output["message"] = "done.";
```

Comment:
process error code,
and take care of the json responce to conform json-rpc-v2

Reply:
after modification:
jv_output = errcode.message();